### ISSUE TITLE
Multiple 'broken' transactions issue

### ISSUE THREAD

**Issue by C7-C5** [0xPolygon/cdk-erigon#1049]
#### System information

network8 / acceptance/2.0.0-beta17-8c3b1fa7706e03295143dca6990806b6745fc3e1

Multiple identical transactions yield different outcomes.
Let's say we conducted 10 transactions.
```
`ttxs:1 round:0 tx:0 sender:0 nonce:0: tx_hash=0xb4e49c0eb87855040e169267e3530a5eb49e521829dde6ef89984f4d2c0f867d | data=...6a606a606a606a93 (16238 bytes (16235))
ttxs:2 round:0 tx:1 sender:0 nonce:1: tx_hash=0xde89278aca504ceb61a8c56ac7bd0098c710be241938c2fdd56b7cedc42dee5e | data=...6d606d606d606d96 (9837 bytes (9837))
ttxs:3 round:0 tx:2 sender:0 nonce:2: tx_hash=0xac3ad61bc126b79160cc2a162fcadf99bb1181350e868c1c99d8a538e9e58213 | data=...766076607660769f (14222 bytes (14196))
ttxs:4 round:0 tx:3 sender:0 nonce:3: tx_hash=0x014a989a232ed0b64a1c023aacc2c35cd0be61644acfc11953dbdc9731681e93 | data=...827e5a6006600606 (8844 bytes (8841))
ttxs:5 round:0 tx:4 sender:0 nonce:4: tx_hash=0x3eb16a6710415793d4a5e7d8c814d2cb4828e3b903732de996d82eef5df92606 | data=...746074607460749d (2005 bytes (1988))
ttxs:6 round:0 tx:5 sender:0 nonce:5: tx_hash=0xd454e0775c52148ee1cfd754d474d9919f0b5b65630384f18cbe0b8aed1cd53c | data=...5f605f605f605f88 (12830 bytes (12826))
ttxs:7 round:0 tx:6 sender:0 nonce:6: tx_hash=0xd5b227b124ebbf95df62a79860b3c5c4d0f61fcd15350bdbede7aa2e7dbf42ac | data=...7d607d607d607df1 (11048 bytes (11042))
ttxs:8 round:0 tx:7 sender:0 nonce:7: tx_hash=0x527fc953bf57fb0c7d7eaf7e1691cb3231e657ebb04e09ef4916fd63ecbdba65 | data=...6008086067606790 (13508 bytes (13506))
ttxs:9 round:0 tx:8 sender:0 nonce:8: tx_hash=0x83bbc660a46991831831974b42e0b850be58620d2c3b363e0639e16cb91d6f72 | data=...746074607460749d (7142 bytes (7120))
ttxs:10 round:0 tx:9 sender:0 nonce:9: tx_hash=0xcf0ce46ac50c81b0a9ff213b73024de9c9b2c196416dfcd2f8b01eea710e883d | data=...6081ff607f607ff3 (970 bytes (968))`
```

User's **nonce** in this case would range from 3 to 6 (while 10 is expected) because some transactions remain unconfirmed.
Moreover 'confirmed' can be traced (for example tx_hash=0xb4e49c0eb87855040e169267e3530a5eb49e521829dde6ef89984f4d2c0f867d)

```
`curl -X POST http://34.175.214.161:18123/ \                       ✔ 
     -H "Content-Type: application/json" \
     -d '{
           "jsonrpc": "2.0",
           "method": "zkevm_traceTransactionCounters",
           "params": ["0xb4e49c0eb87855040e169267e3530a5eb49e521829dde6ef89984f4d2c0f867d"],      
           "id": 1
         }'

{"jsonrpc":"2.0","id":1,"result":{"smtLevels":32,"structLogs":[{"pc":0,"op":"PUSH1","gas":29687395,"gasCost":3,"depth":1,"stack":[],"memory":[],"counters":{"S":0,"D":0,"SHA":0,"A":0,"B":0,"K":0,"M":0,"P":0}},{"pc":2,"op":"PUSH1","gas":29687392,"gasCost":3,"depth":1,"stack":["0x16"],"memory":[],"counters":{"SHA":0,"A":12,"B":30,"K":0,"M":2,"P":0,"S":514,"D":0}},{"pc":4,"op":"BYTE","gas":29687389,"gasCost":3,"depth":1,"stack":["0x16","0x16"],"memory":[],"counters":{"M":4,"P":0,"S":1028,"D":0,"SHA":0,"A":24,"B":60,"K":0}},{"pc":5,"op":"PUSH23","gas":29687386,"gasCost":3,"depth":1,"stack":["0x0"],"memory":[],"counters":{"S":1375,"D":0,"SHA":0,"A":32,"B":82,"K":0,"M":5,"P":0}},{"pc":29,"op":"PUSH20","gas":29687383,"gasCost":3,"depth":1,"stack":["0x0","0x6a60306030556011156d6004600404326c737560578060"],"memory":[],"counters":{"S":1889,"D":0,"SHA":0,"A":44,"B":112,"K":0,"M":7,"P":0}},{"pc":50,"op":"PUSH1","gas":29687380,"gasCost":3,"depth":1,"stack":["0x0","0x6a60306030556011156d6004600404326c737560578060","0x6073607360736073607360736073607360736073"],"memory":[],"counters":{"M":9,"P":0,"S":2403,"D":0,"SHA":0,"A":56,"B":142,"K":0}},{"pc":52,"op":"PUSH1","gas":29687377,"gasCost":3,"depth":1,"stack":["0x0","0x6a60306030556011156d6004600404326c737560578060","0x6073607360736073607360736073607360736073","0x73"],"memory":[],"counters":{"D":0,"SHA":0,"A":68,"B":172,"K":0,"M":11,"P":0,"S":2917}},{"pc":54,"op":"PUSH1","gas":29687374,"gasCost":3,"depth":1,"stack":["0x0","0x6a60306030556011156d6004600404326c737560578060","0x6073607360736073607360736073607360736073","0x73","0x73"],"memory":[],"counters":{"M":13,"P":0,"S":3431,"D":0,"SHA":0,"A":80,"B":202,"K":0}},{"pc":56,"op":"SWAP13","gas":29687371,"gasCost":3,"depth":1,"error":"stack underflow (6 <=> 14)","stack":["0x0","0x6a60306030556011156d6004600404326c737560578060","0x6073607360736073607360736073607360736073","0x73","0x73","0x73"],"memory":[],"counters":{"SHA":0,"A":92,"B":232,"K":0,"M":15,"P":0,"S":3945,"D":0}],"gas":29999999,"failed":true,"returnValue":""}}`

```
whereas "broken" transactions cannot be tracked.
```
`curl -X POST http://34.175.214.161:18123/ \                       ✔ 
     -H "Content-Type: application/json" \
     -d '{
           "jsonrpc": "2.0",
           "method": "zkevm_traceTransactionCounters",
           "params": ["0xcf0ce46ac50c81b0a9ff213b73024de9c9b2c196416dfcd2f8b01eea710e883d"],      
           "id": 1
         }'

{"jsonrpc":"2.0","id":1,"result":null}`
```

Expected behaviour: all transations pass, user's nonce == 10 (for ten txs)
Steps to reproduce:
1. Send n identical transactions
2. Check it by zkevm_traceTransactionCounters or by eth_getTransactionReceipt
3. Ensure the nonce matches / doesn't match with initial nonce + n

NB: This behavior occurs only during 'corrupted' transactions. It's currently impossible to confirm this behavior with 'normal' transactions.

**Comment by Sharonbc01** [comment]
fix in and will be merged into zkEVM branch and will look to merge in stable CDK Erigon branch shortly.

**Comment by mandrigin** [comment]
The issue is fixed in bde1b5ecf931c9b367a7ebc0b1d60c0830f83493, which is back ported to the stable branch.
